ddp-workspace/projects/study-builder-ui/src/app/components/picklist-question-editor/picklist-question-editor.component.ts

```diff
@@ -83,11 +83,18 @@ export class PicklistQuestionEditorComponent implements OnInit, OnDestroy {
         question.validations = formData.required ? [{ ruleType: 'REQUIRED', hintTemplate: null }] : [];
         question.selectMode = formData.selectMode;
         question.renderMode = formData.renderMode;
-        question.picklistOptions = formData.options;
+        question.picklistOptions = formData.optionsData.options;
+        question.groups = formData.optionsData.groups;
         return question;
     }
 
     private currentQuestion(): PicklistQuestionDef | null {
         return this.questionBlockSubject.getValue()?.question as PicklistQuestionDef;
     }
+
+    public allowCreateOptionsGroups(): boolean {
+        const selectMode = this.formGroup.get('selectMode').value;
+        const renderMode = this.formGroup.get('renderMode').value;
+        return selectMode === 'MULTIPLE' && renderMode === 'LIST' || renderMode === 'CHECKBOX_LIST' || renderMode === 'AUTOCOMPLETE';
```

Comment:
_**#minor**_

can be shortened like
```
return selectMode === 'MULTIPLE' && ['LIST', 'CHECKBOX_LIST', 'AUTOCOMPLETE'].includes(renderMode);
```

Reply:
not really, the logical operator will work like 
"(selectMode === 'MULTIPLE' && renderMode === 'LIST') || renderMode === 'CHECKBOX_LIST' || renderMode === 'AUTOCOMPLETE';"

Reply:
My bad, agree.